web/src/components/apps/DashboardVersionCard.tsx

```diff
@@ -732,134 +833,38 @@ class DashboardVersionCard extends React.Component<Props, State> {
-      if (version.yamlErrors) {
-        this.setState({
-          displayShowDetailsModal: !this.state.displayShowDetailsModal,
-          deployView: true,
-          versionToDeploy: version,
-          yamlErrorDetails: version.yamlErrors,
-        });
-        return;
-      }
-      if (version.status === "pending_preflight") {
-        this.setState({
-          showSkipModal: true,
-          versionToDeploy: version,
-          isSkipPreflights: true,
-        });
-        return;
-      }
-      if (version?.preflightResult && version.status === "pending") {
-        const preflightResults = JSON.parse(version.preflightResult);
-        const preflightState = getPreflightResultState(preflightResults);
-        if (preflightState === "fail") {
-          this.setState({
-            showDeployWarningModal: true,
-            versionToDeploy: version,
-          });
-          return;
-        }
-      }
-
-      // prompt to make sure user wants to deploy
-      this.setState({
-        displayConfirmDeploymentModal: true,
-        versionToDeploy: version,
-        isRedeploy: redeploy,
-      });
-      return;
-    } else {
-      // force deploy is set to true so finalize the deployment
-      this.finalizeDeployment(continueWithFailedPreflights, redeploy);
-    }
-  };
-
-  finalizeDeployment = async (
-    continueWithFailedPreflights: boolean,
-    redeploy: boolean
-  ) => {
-    const { match } = this.props;
-    const { versionToDeploy, isSkipPreflights } = this.state;
-    this.setState({ displayConfirmDeploymentModal: false, confirmType: "" });
-    if (redeploy) {
-      await this.props.redeployVersion(match.params.slug, versionToDeploy);
-    }
-    if (versionToDeploy) {
-      await this.props.makeCurrentVersion(
-        match.params.slug,
-        versionToDeploy,
-        isSkipPreflights,
-        continueWithFailedPreflights
-      );
-      this.setState({ versionToDeploy: null, isRedeploy: false });
-
-      if (this.props.refetchData) {
-        this.props.refetchData();
-      }
-    } else {
-      throw new Error("No version to deploy");
-    }
-  };
-
-  onForceDeployClick = (continueWithFailedPreflights = false) => {
-    this.setState({
+  const onForceDeployClick = (continueWithFailedPreflights = false) => {
+    setState({
       showSkipModal: false,
       showDeployWarningModal: false,
       displayShowDetailsModal: false,
     });
-    const versionToDeploy = this.state.versionToDeploy;
+    const versionToDeploy = state.versionToDeploy;
     if (versionToDeploy) {
-      this.deployVersion(versionToDeploy, true, continueWithFailedPreflights);
+      deployVersion(versionToDeploy, true, continueWithFailedPreflights);
     } else {
       throw new Error("No version to deploy");
     }
   };
 
-  showReleaseNotes = (releaseNotes: string) => {
-    this.setState({
-      showReleaseNotes: true,
-      releaseNotes: releaseNotes,
-    });
-  };
-
-  hideReleaseNotes = () => {
-    this.setState({
```

Comment:
as far as i can tell this `showReleaseNotes` is never called so removing for now.